### ISSUE TITLE
GROUP BY cannot access renamed columns

### ISSUE THREAD

**Issue by janetzki** [hyrise/hyrise#1609]
**Bug report**
```sql
SELECT c_custkey AS custid
FROM customer
GROUP BY c_custkey
```
returns
`Invalid input error: Couldn't resolve identifier 'c_custkey' or it is ambiguous`

TPC-DS 1, as generated by the official toolset, requires a column name similar to `c_custkey` to be accessible in the `GROUP BY` clause.

**Comment by mrzzzrm** [comment]
First of all, please check that this is not due to a query template for a specific DB. I am pretty sure I checked other DBs for where in the SELECT statement aliases become visible - which resulted in the current behaviour: Aliases from the SELECT clause are visible nly for GROUP BY and HAVING. See the usage of `SQLIdentifierResolver` - which is the class holding the current set of available identifiers while translating an AST to a LQP - in `SQLTranslator::_translate_select_list_groupby_having`

sqlfiddle.com is a good place to try out different DBs, btw.

If changes to Hyrise's current behaviour are necessary I'd be happy to give you a quick tour of the `SQLTranslator` and how it handles aliases.

**Comment by janetzki** [comment]
Closed by #1693
